jetson/nav/search/searchStateMachine.cpp

```diff
@@ -259,18 +305,56 @@ NavState SearchStateMachine::executeDriveToTarget()
     double distance = mRover->roverStatus().leftTarget().distance;
     double bearing = mRover->roverStatus().leftTarget().bearing + mRover->roverStatus().odometry().bearing_deg;
 
+    // Update if we have an empty leftTarget, BUT we have a valid cached leftTarget
+    if ( mRover->roverStatus().leftTarget().distance < 0 && mRover->roverStatus().leftCacheTarget().distance != -1 ) {
+        distance = mRover->roverStatus().leftCacheTarget().distance;
+        bearing = mRover->roverStatus().leftCacheTarget().bearing + mRover->roverStatus().odometry().bearing_deg;
+        cout << "USING CACHE" << endl;
+    }
+
     // Executes the logic for driving with 0, 1, or 2 targets in sight
     // If we have a second target detected, determine which post is closer
     // If the distance to the second target is less than the first,
     // set our variables to the target 2's distance and bearing
     // Else, use the initialized values from target 1 when driving
     if( mRover->roverStatus().rightTarget().distance > 0 )
     {
+        // Valid leftTarget, perform comp.
         if( mRover->roverStatus().leftTarget().distance > mRover->roverStatus().rightTarget().distance ) 
         {
             distance = mRover->roverStatus().rightTarget().distance;
             bearing = mRover->roverStatus().rightTarget().bearing + mRover->roverStatus().odometry().bearing_deg;
         }
+        // If invalid leftTarget, check Cached target
+        else if( mRover->roverStatus().leftTarget().distance < 0 && mRover->roverStatus().leftCacheTarget().distance != -1 ) 
```

Comment:
Isn't it impossible to have an invalid left target if you have a valid right target? How are the left target and right target determined?

Reply:
Good catch, I wanted to catch all possible cases, but forgot that if leftTarget is null, rightTarget is guaranteed to be null.

Reply:
For the other part though, we still need to check if leftTarget is valid, since rightTarget being invalid does not imply that leftTarget is, if that makes sense?